src/OwnershipRegistry.sol

```diff
@@ -586,13 +625,10 @@ contract OwnershipRegistry is Earnable, OwnableUpgradeable, UUPSUpgradeable {
     ///          V2 changes to require providing Keeper auction royalty to prevent front-running.
     /// @param  orbId      ID of the Orb to purchase.
     /// @param  newPrice_  New price to use after the purchase.
-    function purchase(uint256 orbId, uint256 newPrice_)
-        external
-        payable
-        virtual
-        onlyKeeperHeld(orbId)
-        onlyKeeperSolvent(orbId)
-    {
+    function purchase(uint256 orbId, uint256 newPrice_) external payable virtual onlyKeeperHeld(orbId) {
```

Comment:
Shouldn't this function also revert if there's a standing purchase order? (`purchaseOrder[orbId].purchaser != address(0)`)?

Reply:
Hmm, my thinking was to allow immediate risky purchase, if you don't want to to wait until Orb recharges, and accept the rug risk.

Immediate risky purchase is allowed before a purchase order, and in here it also compensates the purchase order maker. Can't see large pros or cons either way. Maybe will remove this feature to keep it simpler.